### PR TITLE
std::process for fuchsia: updated to latest liblaunchpad

### PR DESCRIPTION
**Pull request by tedsta**
Our liblaunchpad changed a bit and so fuchsia's std::process impl needs to change a bit.

@raphlinus

### CODE REVIEW

File: src/libstd/sys/unix/process/process_fuchsia.rs [hunk 1]
```diff
@@ -51,7 +51,7 @@ impl Command {
     }
 
     unsafe fn do_exec(&mut self, stdio: ChildPipes)
-                      -> io::Result<(*mut launchpad_t, mx_handle_t)> {
+                      -> io::Result< mx_handle_t> {
```
Comment (raphlinus) [inline src/libstd/sys/unix/process/process_fuchsia.rs]: Extra space here?

**Comment by raphlinus** [comment]
This looks good to me, on reading the code. I'll build locally as well, just to make sure.

File: src/libstd/sys/unix/process/process_fuchsia.rs [hunk 2]
```diff
@@ -111,12 +110,16 @@ impl Command {
             callback()?;
         }
 
-        let process_handle = mx_cvt(launchpad_start(launchpad))?;
+        let mut process_handle: mx_handle_t = 0;
+        let mut err_msg: *const libc::c_char = ptr::null();
+        mx_cvt(launchpad_go(launchpad, &mut process_handle, &mut err_msg))?;
+        // TODO: See if we want to do something with that err_msg
```
Comment (raphlinus) [inline src/libstd/sys/unix/process/process_fuchsia.rs]: The "tidy" linter didn't like this. It feels like a TODO to me, but if it wants FIXME, I suppose we should make it happy.

**Comment by alexcrichton** [comment]
Looks great, thanks! r=me with the tidy failure fixed

**Comment by raphlinus** [comment]
Confirm that Rust is functional again, xi editor builds and runs. I haven't deeply exercised process creation though.

**Comment by tedsta** [comment]
@bors r=alexcrichton tidy fixed

@raphlinus yeah I plan to write a bunch of unit tests for sys/fuchsia to give us some peace of mind.

File: src/libstd/sys/unix/process/process_fuchsia.rs (hunk 1 above)
Comment (tedsta) [inline src/libstd/sys/unix/process/process_fuchsia.rs]: thanks for the catch

**Comment by tedsta** [comment]
Looks like some tests failed because of S3 outage yesterday. Is there a way I can trigger a rerun?

**Comment by alexcrichton** [comment]
@tedsta oh this PR is [in the queue](https://buildbot2.rust-lang.org/homu/queue/rust) which is the one that really matters, the Travis status on the PR itself is mostly just advisory currently

**Pull request merged**